From 6282a8559eb9258a5dd7b9bdbfbe6d9c631f9ff6 Mon Sep 17 00:00:00 2001
From: Eldar Abusalimov <eldar.abusalimov@jetbrains.com>
Date: Tue, 24 Oct 2017 20:46:55 +0300
Subject: [PATCH 1/4] build: Disable SOVERSION for liblldb.so

---
diff --git a/source/API/CMakeLists.txt b/source/API/CMakeLists.txt
--- a/source/API/CMakeLists.txt
+++ b/source/API/CMakeLists.txt
@@ -161,10 +161,8 @@ if(LLDB_ENABLE_LUA)
   set_source_files_properties(${lldb_lua_wrapper} PROPERTIES GENERATED ON)
 endif()

-set_target_properties(liblldb
-  PROPERTIES
-  VERSION ${LLDB_VERSION}
-)
+set_property(TARGET liblldb PROPERTY VERSION)
+set_property(TARGET liblldb PROPERTY SOVERSION)

 if (NOT CMAKE_SYSTEM_NAME MATCHES "Windows")
   if (NOT LLDB_EXPORT_ALL_SYMBOLS)
-- 
2.23.0

