From f64a55d5f00f17e16d5fd005d51c54839fda6c63 Mon Sep 17 00:00:00 2001
From: Eldar Abusalimov <eldar.abusalimov@jetbrains.com>
Date: Wed, 11 Oct 2017 12:58:39 +0300
Subject: [PATCH 2/4] Allow Python to find its home when relocated

---
diff --git a/source/Plugins/ScriptInterpreter/Python/ScriptInterpreterPython.cpp b/source/Plugins/ScriptInterpreter/Python/ScriptInterpreterPython.cpp
--- a/source/Plugins/ScriptInterpreter/Python/ScriptInterpreterPython.cpp
+++ b/source/Plugins/ScriptInterpreter/Python/ScriptInterpreterPython.cpp
@@ -159,6 +159,22 @@ private:
     if (g_python_home != nullptr) {
       Py_SetPythonHome(g_python_home);
     }
+#elif PY_MAJOR_VERSION >= 3
+    // Make the program relocatable, that is, compute Python location relative
+    // to the running executable (LLDB or a frontend).
+    const std::string executable = llvm::sys::fs::getMainExecutable("",
+        (void *) &llvm::sys::fs::getMainExecutable);  // any LLVM symbol is OK
+    if (!llvm::sys::fs::exists(executable)) {
+      return;
+    }
+    std::string bindir = llvm::sys::path::parent_path(executable).str();
+    std::string python_prog = bindir + "/python";  // doesn't really matter
+
+    static wchar_t *g_python_prog = Py_DecodeLocale(python_prog.c_str(), NULL);
+    if (g_python_prog == nullptr) {
+      return;
+    }
+    Py_SetProgramName(g_python_prog);
 #endif
   }

-- 
2.23.0

