--- a/source/API/CMakeLists.txt
+++ b/source/API/CMakeLists.txt
@@ -117,10 +117,8 @@ target_link_libraries(liblldb PRIVATE
   lldbPluginScriptInterpreterPython
   )
 
-set_target_properties(liblldb
-  PROPERTIES
-  VERSION ${LLDB_VERSION}
-  )
+set_property(TARGET liblldb PROPERTY VERSION)
+set_property(TARGET liblldb PROPERTY SOVERSION)
 
 if (NOT CMAKE_SYSTEM_NAME MATCHES "Windows")
   if (NOT LLDB_EXPORT_ALL_SYMBOLS)
