diff --git a/gdb/configure b/gdb/configure
index 1531f62f76a..44cce598121 100755
--- a/gdb/configure
+++ b/gdb/configure
@@ -28178,7 +28178,7 @@ fi
   # Otherwise fallback on the old way of trying different versions of
   # python in turn.
 
-  have_libpython=no
+  have_libpython=yes
   if test "${have_python_config}" = yes; then
 
 
diff --git a/gdb/python/python.c b/gdb/python/python.c
index acd80e5515c..2d97c38abbd 100644
--- a/gdb/python/python.c
+++ b/gdb/python/python.c
@@ -2345,6 +2345,7 @@ py_initialize_catch_abort ()
 
   TRY_SJLJ
     {
+      PyImport_AppendInittab ("_gdb", init__gdb_module);
       Py_Initialize ();
       py_isinitialized = true;
     }
