diff -Naur a/gdb/cp-valprint.c b/gdb/cp-valprint.c
--- a/gdb/cp-valprint.c	2020-06-10 15:46:03.506264764 +0300
+++ b/gdb/cp-valprint.c	2020-06-10 15:40:43.973949676 +0300
@@ -622,6 +622,12 @@
       return;
     }
 
+  if (recurse > 99)
+  {
+    fputs_filtered ("<recursion limit reached>", stream);
+    return;
+  }
+
   struct type *real_type = check_typedef (type);
   if (TYPE_CODE (real_type) == TYPE_CODE_STRUCT)
     {
