diff --git a/src/output.h b/src/output.h
index a506505..28a97ec 100644
--- a/src/output.h
+++ b/src/output.h
@@ -67,13 +67,21 @@ void output_dump (struct output *out);
 
 # ifdef WINDOWS32
 /* For emulations in w32/compat/posixfcn.c.  */
-#  define F_GETFD 1
-#  define F_SETLKW 2
+#  ifndef F_GETFD
+#   define F_GETFD 1
+#  endif
+#  ifndef F_SETLKW
+#   define F_SETLKW 2
+#  endif
 /* Implementation note: None of the values of l_type below can be zero
    -- they are compared with a static instance of the struct, so zero
    means unknown/invalid, see w32/compat/posixfcn.c. */
-#  define F_WRLCK 1
-#  define F_UNLCK 2
+#  ifndef F_WRLCK
+#   define F_WRLCK 1
+#  endif
+#  ifndef F_UNLCK
+#   define F_UNLCK 2
+#  endif
 
 struct flock
   {
@@ -89,7 +97,7 @@ struct flock
 typedef intptr_t sync_handle_t;
 
 /* Public functions emulated/provided in posixfcn.c.  */
-int fcntl (intptr_t fd, int cmd, ...);
+//int fcntl (intptr_t fd, int cmd, ...);
 intptr_t create_mutex (void);
 int same_stream (FILE *f1, FILE *f2);
 
diff --git a/src/w32/compat/posixfcn.c b/src/w32/compat/posixfcn.c
index 975dfb7..d8cb8d9 100644
--- a/src/w32/compat/posixfcn.c
+++ b/src/w32/compat/posixfcn.c
@@ -30,6 +30,7 @@ this program.  If not, see <http://www.gnu.org/licenses/>.  */
 /* Support for OUTPUT_SYNC and related functionality.  */
 
 /* Emulation of fcntl that supports only F_GETFD and F_SETLKW.  */
+#if 0
 int
 fcntl (intptr_t fd, int cmd, ...)
 {
@@ -142,6 +143,7 @@ fcntl (intptr_t fd, int cmd, ...)
         return -1;
     }
 }
+#endif
 
 static intptr_t mutex_handle = -1;
 
